pkg/reconciler/jbsconfig/jbsconfig.go

```diff
@@ -634,3 +695,110 @@ func (r *ReconcilerJBSConfig) processSpiImageSecret(ctx context.Context, config
 	}
 	return errors2.New("created SPIAccessTokenBinding, waiting for secret to be injected")
 }
+
+func (r *ReconcilerJBSConfig) handleNoOwnerSpecified(ctx context.Context, log logr.Logger, config *v1alpha1.JBSConfig) error {
+	if r.quayClient == nil || r.quayOrgName == "" {
+		return errors2.New("no owner specified and automatic repo creation is disabled")
+	}
+
+	repo, robotAccount, err := r.generateImageRepository(log, config)
+	if err != nil {
+		return err
+	}
+	if repo == nil || robotAccount == nil {
+		return errors2.New("unknown error in the repository generation process")
+	}
+	controllerutil.AddFinalizer(config, ImageRepositoryFinalizer)
+	config.Spec.Owner = r.quayOrgName
```

Comment:
TODO before merging: create Status fields for these so this is not updating the spec.